src/main.rs

```diff
@@ -44,39 +53,108 @@ impl<'a> rocket::request::FromParam<'a> for CidWrap {
     }
 }
 
-struct Store<T>
+struct Orbits<O>
 where
-    T: ContentAddressedStorage,
+    O: Orbit,
 {
-    pub db: T,
+    pub stores: RwLock<BTreeMap<Cid, O>>,
+    pub base_path: PathBuf,
+}
+
+#[rocket::async_trait]
+pub trait OrbitCollection<O: Orbit> {
+    async fn orbits(&self) -> RwLockReadGuard<BTreeMap<Cid, O>>;
+    async fn add(&self, orbit: O) -> ();
+}
+
+impl<O: Orbit> Orbits<O> {
+    fn new<P: AsRef<Path>>(path: P) -> Self {
+        Self {
+            stores: RwLock::new(BTreeMap::new()),
+            base_path: path.as_ref().to_path_buf(),
+        }
+    }
+}
+
+#[rocket::async_trait]
+impl<O: Orbit> OrbitCollection<O> for Orbits<O> {
+    async fn orbits(&self) -> RwLockReadGuard<BTreeMap<Cid, O>> {
+        self.stores.read().await
+    }
+
+    // fn orbit(&self, id: &Cid) -> Option<&O> {
+    //     self.stores.read().expect("read orbit set").get(id)
+    // }
+
+    async fn add(&self, orbit: O) {
+        let mut lock = self.stores.write().await;
+        lock.insert(*orbit.id(), orbit);
+    }
+}
+
+async fn load_orbits<P: AsRef<Path>>(
```

Comment:
You don't need a generic for these cases, you can simply have `path: dyn AsRef<Path>`. Same for `Orbits` I believe

Reply:
good point